src/Seq.Api/Api/SeqConnection.cs

```diff
@@ -11,13 +11,16 @@ namespace Seq.Api
     public class SeqConnection : ISeqConnection
     {
         readonly SeqApiClient _client;
-        readonly Dictionary<string, ResourceGroup> _resourceGroups = new Dictionary<string, ResourceGroup>();
-        RootEntity _root;
+        readonly ConcurrentDictionary<string, Task<ResourceGroup>> _resourceGroups = new ConcurrentDictionary<string, Task<ResourceGroup>>();
+        readonly Lazy<Task<RootEntity>> _root;
 
         public SeqConnection(string serverUrl, string apiKey = null)
         {
             if (serverUrl == null) throw new ArgumentNullException("serverUrl");
             _client = new SeqApiClient(serverUrl, apiKey);
+            
+            _root = new Lazy<Task<RootEntity>>(() => _client.GetRootAsync());
```

Comment:
If this is concurrent, LazyThreadSafetyMode will need to be specified on the constructor.